Electron/RON/components/PingMap.tsx

```diff
@@ -0,0 +1,290 @@
+    ]
+    for (let i = 0; i < locations.length && i < boards.length; i++) {
+      const coords = locations[i]
+      const board = boards[i]
+      context.fillStyle =
+        this.state.ping[board] === -1
+          ? "white"
+          : ColorConverter(this.state.ping[board], min, cutoff, max, greenHue, redHue)
+      context.beginPath()
+      context.moveTo(centerW, 400)
+      context.lineTo(coords.x, coords.y)
+      context.stroke()
+      context.moveTo(20 * Math.cos((1 / 6) * Math.PI) + coords.x, 20 * Math.sin((1 / 6) * Math.PI) + coords.y)
+      context.lineTo(20 * Math.cos((5 / 6) * Math.PI) + coords.x, 20 * Math.sin((5 / 6) * Math.PI) + coords.y)
+      context.lineTo(20 * Math.cos((9 / 6) * Math.PI) + coords.x, 20 * Math.sin((9 / 6) * Math.PI) + coords.y)
+      context.lineTo(20 * Math.cos((1 / 6) * Math.PI) + coords.x, 20 * Math.sin((1 / 6) * Math.PI) + coords.y)
+      context.stroke()
+      context.fill()
+      context.fillStyle = "black"
+      context.textAlign = "center"
+      context.fillText(board, coords.x, coords.y + 17)
+    }
+    context.beginPath()
+    context.fillStyle =
+      this.state.ping.GrandStream === -1
+        ? "white"
+        : ColorConverter(this.state.ping.GrandStream, min, cutoff, max, greenHue, redHue)
+    context.rect(centerW - 20, 380, 40, 40)
+    context.stroke()
+    context.fill()
+    text = "Router"
+    context.textBaseline = "middle"
+    context.fillStyle = "black"
+    context.textAlign = "center"
+    context.fillText(text, centerW, 400)
+  }
+
+  ICMP(device: string): void {
+    // If device is not a network device, it must be a board
+    let deviceInfo = NetworkDevices[device]
+    if (deviceInfo === undefined) {
+      deviceInfo = DATAID[device]
+    }
+    const ip = deviceInfo.Ip
+
+    // Ping command is slightly different for windows/unix
+    let pingCommand = ""
+    if (process.platform === "win32") {
+      pingCommand = `ping -n 1 ${ip}`
+    } else {
+      pingCommand = `ping -c 1 ${ip}`
+    }
+
+    exec(pingCommand, (error, stdout, stderr) => {
+      // -1 means not reachable
+      let delay = -1
+      if (!error && !stderr && stdout.indexOf("unreachable") === -1) {
+        // On Windows, failed ping is not necessarily an error
+        // Windows Ex. 'Reply from 8.8.8.8: bytes=32 time=26ms TTL=110'
+        // Unix Ex. '64 bytes from 8.8.8.8: icmp_seq=0 ttl=110 time=387.477 ms'
+        const start = stdout.indexOf("time=") + 5
+        const end = stdout.indexOf("ms")
+        delay = Math.round(parseFloat(stdout.substring(start, end)))
+      }
+      this.setState({
+        ping: {
+          ...this.state.ping,
+          [device]: delay,
+        },
+      })
+    })
+  }
+
+  StartAutoPing(interval: number): void {
+    setInterval(() => {
+      for (const device in this.state.ping) {
+        if (Object.prototype.hasOwnProperty.call(this.state.ping, device)) {
+          this.ICMP(device)
+        }
+      }
+      this.updatePingMap()
```

Comment:
Eventually, it would be nice to look into a better way of doing the pingmap. However, with the way the paths are drawn and the colors filled, it has to be completely redrawn in order to update the fills. The priority of this improvement will depend heavily on how well/poorly the application performs when connected to the rover.